src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkInterface.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net.NetworkInformation
+{
+    public abstract class NetworkInterface
+    {
+        /// Returns objects that describe the network interfaces on the local computer.
+        public static NetworkInterface[] GetAllNetworkInterfaces()
+        {
+            return SystemNetworkInterface.GetNetworkInterfaces();
+        }
+
+        public static bool GetIsNetworkAvailable()
+        {
+            return SystemNetworkInterface.InternalGetIsNetworkAvailable();
+        }
+
+        public static int LoopbackInterfaceIndex
+        {
+            get
+            {
+                return SystemNetworkInterface.InternalLoopbackInterfaceIndex;
+            }
+        }
+
+        public static int IPv6LoopbackInterfaceIndex
+        {
+            get
+            {
+                return SystemNetworkInterface.InternalIPv6LoopbackInterfaceIndex;
+            }
+        }
+
+        public virtual string Id { get { throw NotImplemented.ByDesignWithMessage(SR.net_PropertyNotImplementedException); } }
```

Comment:
I agree the design is not ideal. Unfortunately this is part of the public contract. 
@mellinoe Throwing PlatformNotSupported for the implementations sounds good.

Won't fix for app-compat reasons.